### PR TITLE
[restclient-cpp] new port

### PR DESCRIPTION
**Pull request by et-nik**
This adds the C++ REST Client Library https://github.com/mrtazz/restclient-cpp/tree/0.5.0

### CODE REVIEW

**Comment by vicroms** [comment]
Hi @et-nik 

Thanks for contributing to vcpkg!

Looks like CMake is not oficially supported by `restclient-cpp`. 
It would be great if you can submit a PR upstream with your proposed CMake integration.

File: ports/restclient-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,162 @@
+
```
Comment (vicroms) [inline ports/restclient-cpp/CMakeLists.txt]: Why not use the CMakeLists.txt shipped with the library?

https://github.com/mrtazz/restclient-cpp/blob/master/CMakeLists.txt

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/restclient-cpp/CMakeLists.txt (hunk 1 above)
Comment (et-nik) [reply ports/restclient-cpp/CMakeLists.txt]: I used this CMakeLists.txt, but I changed one line to build a static library.

https://github.com/mrtazz/restclient-cpp/blob/master/CMakeLists.txt#L42

I change to 
```
add_library(restclient-cpp STATIC
  source/restclient.cc
  source/connection.cc
  source/helpers.cc
)
```

Comment (vicroms) [reply ports/restclient-cpp/CMakeLists.txt]: You don't need to change the CMakeLists.txt file to build static libraries with vcpkg.

Using `./vcpkg install restclient-cpp:x64-windows-static` or `./vcpkg install restclient-cpp:x86-windows-static` will set `BUILD_SHARED_LIBS` to off and build them static instead.

Comment (et-nik) [reply ports/restclient-cpp/CMakeLists.txt]: I add last release version 0.5.0 but it version has not CMakeFile.txt. This file created in later commit.

What should I do?
I can replace from 0.5.0 version to latest commit version (with CMakeFile.txt).

Comment (vicroms) [reply ports/restclient-cpp/CMakeLists.txt]: Yes, using the full SHA of the latest commit should work.

**Comment by grdowns** [comment]
Hey @et-nik, thanks for the port contribution. I'm going to close this PR due to inactivity, but feel free reopen it with more commits; we'd be happy to give any help we can to continue the work

**Comment by et-nik** [comment]
Hey @grdowns, I added a new commit. 
Updated restclient-cpp version.

**Pull request closed without merging**